### DISCUSSION TITLE
Manage an 'account-exists-with-different-credential' AuthFailed exception?

### DISCUSSION THREAD

**Discussion by jeanmatthieud** [firebase/FirebaseUI-Flutter#79] [Q&A] [answered]
I want to manage an 'account-exists-with-different-credential' error during sign-in.

To produce this error (on Android):
- The user have the address myemail@gmail.com
- The uses this email address for a Facebook account (logged in in the Facebook app)
- The user uses an app with a SignInScreen, with two providers: Google and Facebook
- The user creates an account in this app with the Google provider
- The user log out
- The user tries to log in with the Facebook provider

The callback `AuthStateChangeAction<AuthFailed>` is called, with a `FirebaseAuthException`, with the error code `account-exists-with-different-credential`.

The fields `email` AND `credentials` of the `FirebaseAuthException` are empty.  
The [official Firebase documentation](https://firebase.google.com/docs/auth/flutter/errors?hl=en#handling_account-exists-with-different-credential_errors) explains that we have to use the email address and the credential. But those data are not available in the AuthFailed callback:
![image](https://github.com/firebase/FirebaseUI-Flutter/assets/1443167/d8d62c98-252a-409a-a2d1-fef81055a993)

I don't know if it's a Firebase bug or a FirebaseUi issue.

I'm using the latest package versions:
- firebase_ui_auth: ^1.6.5
- firebase_ui_localizations: ^1.5.3
- flutter_facebook_auth: ^6.0.1
- google_sign_in: ^6.1.4
- firebase_ui_oauth_apple: ^1.2.9
- firebase_ui_oauth_facebook: ^1.2.9
- firebase_ui_oauth_google: ^1.2.9

**Comment by lesnitsky** [comment, marked as answer]
You can handle this using `AuthStateChangeAction<DifferentSignInMethodsFound>`.

```dart
AuthStateChangeAction<DifferentSignInMethodsFound>((context, state) {
  final methods = state.methods; // an array of all provider id's associated with the email
  // your custom logic here
});
```

If you're using `SignInScreen`, you can pass this actions to `actions` list.

If you're building more customized UI – wrap your widget tree with `FirebaseUIAction`:

```dart
FirebaseUIActions(
  actions: [
    AuthStateChangeAction<DifferentSignInMethodsFound>(handler),
  ],
  child: yourWidget,
);
```

There's also [`showDifferentMethodSignInDialog`](https://pub.dev/documentation/firebase_ui_auth/latest/firebase_ui_auth/showDifferentMethodSignInDialog.html) that will show relevant UI for given providers, so it could be used to handle this scenario.
